### DISCUSSION TITLE
Lesson 9 - Private key in vm.startBroadcast()

### DISCUSSION THREAD

**Discussion by ericywl** [Cyfrin/foundry-full-course-cu#1733] [Q&A]
Given that `cast` now supports wallet imports: https://github.com/foundry-rs/foundry/issues/6034, is it possible to get the wallet private key using forge instead of putting them in the `.env` file? For example, via some cheatcode that enables us to get the wallet and obtain the private key?

For reference, deploying via `forge script` already supports something like it via:
```sh
$ forge script script/DeploySimpleStorage.s.sol --rpc-url $SEPOLIA_RPC_URL --broadcast --account <some_account>
```

But `forge test` does not. Maybe something like the following would be great:
```sh
$ forge test --rpc-url $SEPOLIA_RPC_URL --account <some_account>
```
And then in tests
```sol
vm.getAccount().privateKey
```

**Comment by EngrPips** [comment]
Interesting, Maybe you can make that suggestion on `Foundry` official Github.

**Reply by Ikpong-Joseph** [reply]
Try running with `forge test -f $SEPOLIA`. Although I had the path to my keystore in my `.env`. I don't think it was used since I wasn't prompted for a password like when I tried running a Sepolia deploy script.

**Reply by EngrPips** [reply]
> Try running with `forge test -f $SEPOLIA`. Although I had the path to my keystore in my `.env`. I don't think it was used since I wasn't prompted for a password like when I tried running a Sepolia deploy script.

Wouldn't this read the private key directly from the environment variable instead of reading it from the path to the keystore?

**Comment by Ikpong-Joseph** [comment]
The raw private key isn't in the .env. Only the path to its file in the
keystore is. Try it.

**Reply by EngrPips** [reply]
That is interesting. I will try it and let you know how it goes.

**Comment by Ikpong-Joseph** [comment]
Great. Let me know your experiences. I'm @Joseph__hpesoJ on X. Let's
connect.
